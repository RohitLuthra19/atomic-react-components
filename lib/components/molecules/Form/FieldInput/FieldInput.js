// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Input from '../../../atoms/Input';
import FieldError from '../../../atoms/FieldError';
import Label from '../../../atoms/Label';
import styles from './FieldInput.style';


type Props = {
  id: string,
  className ?: string,
  type ?: string,
  name: string,
  label: Node,
  labelProps ?: Object,
  fieldProps ?: Object,
  value ?: string | number,
  placeholder ?: string,
};

const FieldInput = ({
  id,
  className,
  type,
  name,
  label,
  labelProps,
  fieldProps,
  ...others
}: Props): Node => (
  // TODO: Add support for label position
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>{label}</Label>
    <Field
      id={id}
      name={name}
      type={type}
      render={({ field }) => <Input type={type} {...others} {...field} />}
      {...fieldProps}
    />
    <ErrorMessage name={name} component={FieldError} />
  </div>

);

FieldInput.defaultProps = {
  placeholder: '',
  className: '',
  type: 'text',
  labelProps: {},
  value: '',
  fieldProps: {},
};

export default styled(FieldInput)`${styles}`;
export { FieldInput as FieldInputVanilla };
