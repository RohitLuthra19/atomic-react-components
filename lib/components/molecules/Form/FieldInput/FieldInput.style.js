import { css } from 'styled-components';

// TODO: Fix the duplicate styling in input.style.js
const styles = css`
  margin-bottom: 0.75rem;

  label {
    display: block;
    margin-bottom: 0.25rem;
  }

  input {
    margin-bottom: 0.25rem;
  }
`;

export default styles;
