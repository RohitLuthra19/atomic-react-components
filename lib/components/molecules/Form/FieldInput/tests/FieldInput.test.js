import React from 'react';
import { shallow, render } from 'enzyme';
import { FieldInputVanilla } from '../FieldInput';

describe('FieldInput Component', () => {
  let FieldInput = '';

  test('should render correctly', () => {
    FieldInput = shallow(<FieldInputVanilla />);
    expect(FieldInput).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Enter username',
    };
    FieldInput = render(<FieldInputVanilla {...props} />);
    expect(FieldInput.find('label').text()).toEqual('Enter username');
  });

  test('should render Input component with proper type', () => {
    const props = {
      name: 'email',
      type: 'email',
      id: 'email',
    };
    FieldInput = render(<FieldInputVanilla {...props} />);
    expect(FieldInput.find('input[type=email]')).toHaveLength(1);
  });
});
