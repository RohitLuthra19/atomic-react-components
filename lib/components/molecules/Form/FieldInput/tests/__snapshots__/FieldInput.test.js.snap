// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<div
  className=""
>
  <Label
    className=""
  />
  <C
    render={[Function]}
    type="text"
  />
  <C
    component={[Function]}
  />
</div>
`;
